File: C++/Array/buyLowSellHigh.cpp
//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
//Best time to Buy and Sell Stock
//Condition: Buy low, sell high
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int buy, profit, maxP = 0;
        if(prices.empty())
        {
            return 0;
        }

        for(int sell=1; sell<prices.size(); sell++)
        {
            if(buy<sell)
            {
                profit = prices[sell] - prices[buy];
                maxP = std::max(maxP, profit);
            }
            else
            {
                buy = sell;
            }
        }
    return maxP;
    }
};

File: python/Array/buyLowSellHigh.py
#https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
#Best Time to Buy and Sell Stock
#Condition: Buy low, sell high
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy, sell = 0, 1 
        maxP = 0

        while sell < len(prices):
            #check profitable?
            if prices[buy] < prices[sell]:
                profit = prices[sell] - prices[buy]
                maxP = max(maxP, profit)
            else:
                buy = sell
            sell+=1
        return maxP
